pkg/server/config.go

```diff
@@ -6,156 +6,63 @@ import (
 	InstallWorkspaceController bool
 	KubeConfigPath             string
 	Listen                     string
-	PullMode                   bool
-	PushMode                   bool
-	ResourcesToSync            []string
 	RootDirectory              string
-	SyncerImage                string
 	ProfilerAddress            string
 	ShardKubeconfigFile        string
 	EnableSharding             bool
 }
 
-// DefaultConfig returns a configuration with default values.
-func DefaultConfig() *Config {
-	return &Config{
-		AutoPublishAPIs:          false,
-		EtcdClientInfo:           etcd.ClientInfo{},
-		EtcdDirectory:            "data",
-		EtcdPeerPort:             "2380",
-		EtcdClientPort:           "2379",
-		InstallClusterController: false,
-		KubeConfigPath:           "admin.kubeconfig",
-		Listen:                   ":6443",
-		PullMode:                 false,
-		PushMode:                 false,
-		ResourcesToSync:          []string{"deployments.apps"},
-		RootDirectory:            ".kcp",
-		SyncerImage:              "quay.io/kcp-dev/kcp-syncer",
-	}
-}
+func BindOptions(c *Config, fs *pflag.FlagSet) *Config {
+	fs.AddFlag(pflag.PFlagFromGoFlag(flag.CommandLine.Lookup("v")))
+	fs.BoolVar(&c.InstallClusterController, "install_cluster_controller", false, "Registers the sample cluster custom resource, and the related controller to allow registering physical clusters")
+	fs.BoolVar(&c.InstallWorkspaceController, "install_workspace_controller", false, "Registers the workspace custom resource, and the related controller to allow scheduling workspaces to shards")
+	fs.StringVar(&c.Listen, "listen", ":6443", "Address:port to bind to")
+	fs.StringSliceVar(&c.EtcdClientInfo.Endpoints, "etcd-servers", []string{}, "List of external etcd servers to connect with (scheme://ip:port), comma separated. If absent an in-process etcd will be created.")
+	fs.StringVar(&c.EtcdClientInfo.KeyFile, "etcd-keyfile", "", "TLS key file used to secure etcd communication.")
+	fs.StringVar(&c.EtcdClientInfo.CertFile, "etcd-certfile", "", "TLS certification file used to secure etcd communication.")
+	fs.StringVar(&c.EtcdClientInfo.TrustedCAFile, "etcd-cafile", "", "TLS Certificate Authority file used to secure etcd communication.")
+	fs.StringVar(&c.ProfilerAddress, "profiler-address", "", "[Address]:port to bind the profiler to.")
+	fs.StringVar(&c.ShardKubeconfigFile, "shard-kubeconfig-file", "", "Kubeconfig holding admin(!) credentials to peer kcp shards.")
+	fs.BoolVar(&c.EnableSharding, "enable-sharding", false, "Enable delegating to peer kcp shards.")
+	fs.StringVar(&c.RootDirectory, "root_directory", ".kcp", "Root directory.")
+	fs.StringVar(&c.EtcdPeerPort, "etcd_peer_port", "2380", "Port for etcd peer communication.")
+	fs.StringVar(&c.EtcdClientPort, "etcd_client_port", "2379", "Port for etcd client communication.")
+	fs.StringVar(&c.KubeConfigPath, "kubeconfig_path", "admin.kubeconfig", "Path to which the administrative kubeconfig should be written at startup.")
```

Comment:
We want to update the defaults to come from `c`, right?

Reply:
:facepalm: